internal/commands/package_buildpack.go

```diff
@@ -71,11 +68,27 @@ func PackageBuildpack(logger logging.Logger, client BuildpackPackager, packageCo
 		}),
 	}
 	cmd.Flags().StringVarP(&flags.PackageTomlPath, "package-config", "p", "", "Path to package TOML config (required)")
-	cmd.MarkFlagRequired("package-config")
+	cmd.Flags().StringVarP(&flags.PackageTomlPath, "config", "c", "", "Path to package TOML config (required)")
+
+	// TODO: Mark config required and remove package-config after release of pack v0.12
+	// cmd.MarkFlagRequired("config")
+	cmd.Flags().MarkDeprecated("package-config", "please use --config instead\n")
```

Comment:
1. What's the purpose/need for `\n`? If it's solely to provide an additional line break, I would suggest it be removed.
2. Looks like this automatically provides a message. We'd want to make the warning use the `logger.Warn` style. I'm not sure if it's possible to "intercept" and style the output of deprecations in cobra but if not we might want to manually add the styling here to be consistent with prior deprecations.

Reply:
I tried adding the `style.Warn` to this string here; the issue is that `MarkDeprecated` prints a string before the message provided, like so:
```
Flag --builder-config has been deprecated, please use --config instead
```
so only `please use --config instead` had the styling, and it looked pretty odd. I'll look and see whether you can intercept the logs. 

I had added the line break to make the deprecation notice apparent (instead of getting swallowed by the rest of the logs, having a clear break), but I can remove it.

Reply:
Instead of using `MarkDeprecated`, I could use `cmd.Flags().Changed(...)` to detect if it's used, and then print the warning ourselves. https://github.com/spf13/cobra/issues/453